litellm/main.py

```diff
@@ -651,6 +652,38 @@ def completion(
                 response = CustomStreamWrapper(model_response, model, custom_llm_provider="cohere", logging_obj=logging)
                 return response
             response = model_response
+        elif (
+            custom_llm_provider == "gradient"
+            or litellm.api_base == "https://api.gradient.ai/api"
+        ):
+            custom_llm_provider = "gradient"
+            gradient_key = (
+                api_key or litellm.gradient_key or os.environ.get("GRADIENT_API_KEY") or os.environ.get("GRADIENT_ACCESS_TOKEN") or litellm.api_key
+            )
+            gradient_workspace_id = (
+                api_key or litellm.gradient_workspace_id or os.environ.get("GRADIENT_WORKSPACE_ID")
+            )
+
+            model_response = gradient.completion(
+                model=model,
+                messages=messages,
+                model_response=model_response,
+                print_verbose=print_verbose,
+                optional_params=optional_params,
+                litellm_params=litellm_params,
+                logger_fn=logger_fn,
+                encoding=encoding, 
+                api_key=gradient_key, 
+                logging_obj=logging,
+                gradient_workspace_id=gradient_workspace_id
+            )
+            if inspect.isgenerator(model_response) or ("stream" in optional_params and optional_params["stream"] == True):
```

Comment:
for providers that don't support streaming we usually fake streaming. Can you implement it in this PR please ? 

For all our providers we need to support regular completion calls + streaming(if they don't do streaming then we should fake streaming) + Exception mapping

Reply:
you can check out the Fake streaming implementation we did for sagemaker & palm for this